src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCertificateUtils.java

```diff
@@ -0,0 +1,274 @@
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPrivateCrtKeySpec;
+import java.util.Arrays;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+
+import org.bouncycastle.openssl.PEMDecryptorProvider;
+import org.bouncycastle.openssl.PEMEncryptedKeyPair;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder;
+
+
+final class SQLServerCertificateUtils {
+
+    static KeyManager[] getKeyManagerFromFile(String certPath, String keyPath,
+            String keyPassword) throws IOException, GeneralSecurityException, SQLServerException {
+        if (keyPath != null && keyPath.length() > 0) {
+            return readPKCS8Certificate(certPath, keyPath, keyPassword);
+        } else {
+            return readPKCS12Certificate(certPath, keyPassword);
+        }
+    }
+
+    private static KeyManager[] readPKCS12Certificate(String certPath,
+            String keyPassword) throws NoSuchAlgorithmException, CertificateException, FileNotFoundException, IOException, UnrecoverableKeyException, KeyStoreException {
+        KeyStore keystore = KeyStore.getInstance("PKCS12");
+        keystore.load(new FileInputStream(certPath), keyPassword.toCharArray());
+        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+        keyManagerFactory.init(keystore, keyPassword.toCharArray());
+        return keyManagerFactory.getKeyManagers();
+    }
+
+    private static KeyManager[] readPKCS8Certificate(String certPath, String keyPath,
+            String keyPassword) throws IOException, GeneralSecurityException, SQLServerException {
+        Certificate clientCertificate = loadCertificate(certPath);
+        PrivateKey privateKey = loadPrivateKey(keyPath, keyPassword);
+
+        KeyStore keyStore = KeyStore.getInstance("JKS");
+        keyStore.load(null, null);
+        keyStore.setCertificateEntry("client-cert", clientCertificate);
+        keyStore.setKeyEntry("client-key", privateKey, keyPassword.toCharArray(),

```

Comment:
are these hardcoded aliases defined in the specs? can we define them somewhere else?